Rx/v2/src/rxcpp/schedulers/rx-runloop.hpp

```diff
@@ -79,6 +80,9 @@ struct run_loop_scheduler : public scheduler_interface
                 std::unique_lock<std::mutex> guard(st->lock);
                 st->q.push(detail::run_loop_state::item_type(when, scbl));
                 st->r.reset(false);
+                std::function<void()> callback{st->add_task_callback};
+                guard.unlock(); // So we can't get attempt to recursively lock the state
+                if (callback) callback();
```

Comment:
`schedule()` is called from other threads in order to schedule work on the thread that owns the `runloop`. 

`callback()` will be called from other threads. It is not okay to dispatch any events from the callback since they would be running on the wrong thread. The only thing that the callback can do is reset the timer to an earlier value if the new item is supposed to occur earlier than the previous timer was set. This would be easier if the callback was passed the new time.

If so, perhaps it would be better if the callback was only called when the new `when` is earlier than the previous queue top. In that case, the name of the callback might be `notify_next_wakeup` instead of `add_task_callback`.